p023.java

```diff
@@ -0,0 +1,59 @@
+// A perfect number is a number for which the sum of its proper divisors is
+// exactly equal to the number. For example, the sum of the proper divisors of
+// 28 would be 1 + 2 + 4 + 7 + 14 = 28, which means that 28 is a perfect number.
+//
+// A number n is called deficient if the sum of its proper divisors is less
+// than n and it is called abundant if this sum exceeds n.
+//
+// As 12 is the smallest abundant number, 1 + 2 + 3 + 4 + 6 = 16, the smallest
+// number that can be written as the sum of two abundant numbers is 24. By
+// mathematical analysis, it can be shown that all integers greater than 28123
+// can be written as the sum of two abundant numbers. However, this upper
+// limit cannot be reduced any further by analysis even though it is known that
+// the greatest number that cannot be expressed as the sum of two abundant
+// numbers is less than this limit.
+//
+// Find the sum of all the positive integers which cannot be written as the sum
+// of two abundant numbers.
+
+public class p023 {
+    public static void main(String args[]) {
+        System.out.println(sum());
+    }
+
+    private static int LIMIT = 28123;
+
+    private static int sum() {
+        int sum = 0;
+        for (int i = 1; i < LIMIT; i++) {
+            if (!isSumOf2Abundants(i)) {
+                sum += i;
+            }
+        }
+        return sum;
+    }
+
+    private static boolean isSumOf2Abundants(int n) {
+        for (int i = 1; i < n; i++) {
+            if (isAbundant(i) && isAbundant(n - i)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean isAbundant(int n) {
```

Comment:
This function will be called with same parameter for many time on small values. Please avoid this.